Tsegismont shared http client (#4172)
eclipse-vertx/vert.x: src/main/asciidoc/http.adoc

Alternatively you can set idle timeout using {@link io.vertx.core.http.HttpClientOptions#setIdleTimeout(int)} - any
connections not used within this timeout will be closed. Please note the idle timeout value is in seconds not milliseconds.

=== HTTP client sharing

You can share an HTTP client between multiple verticles or instances of the same verticle. Such client should be created outside
of a verticle otherwise it will be closed when the verticle that created it is undeployed

[source,$lang]
----
{@link examples.HTTPExamples#httpClientSharing1}
----

You can also create a shared HTTP client in each verticle:

[source,$lang]
----
{@link examples.HTTPExamples#httpClientSharing2}
----

The first time {@link io.vertx.core.Vertx#createSharedHttpClient(java.lang.String)} is called it will create the

> client. Subsequent calls will reuse this client and create a lease to this client. The client exists until all leases are closed.

Reviewer: ```suggestion
client. Subsequent calls will reuse this client and create a lease to this client. The client is closed after all leases have been closed.
```
